overlord/snapstate/autorefresh_gating.go

```diff
@@ -250,6 +250,39 @@ func ProceedWithRefresh(st *state.State, gatingSnap string) error {
 	return nil
 }
 
+// pruneGating removes affecting snaps that are not in candidates (meaning
+// there is no update for them anymore).
+func pruneGating(st *state.State, candidates map[string]*refreshCandidate) error {
+	gating, err := refreshGating(st)
+	if err != nil {
+		return err
+	}
+
+	if len(gating) == 0 {
+		return nil
+	}
+
+	var changed bool
+	for affectingSnap, gatingSnaps := range gating {
+		if candidates[affectingSnap] == nil {
+			// the snap doesn't have an update anymore, forget it
+			delete(gating, affectingSnap)
+			changed = true
+			continue
+		}
+
+		// after deleting gating snaps we may end up with empty map under
+		// gating[affectingSnap], so remove it.
+		if len(gatingSnaps) == 0 {
+			delete(gating, affectingSnap)
```

Comment:
shouldn't changed be set to true after this delete?

Reply:
No, but uh, it was slightly silly anyway, this condition was useless anyway because we just drop from top-level gating map... It was an artifact of old implementation. Updated. Thanks!